examples/backend/src/ExampleAPI.hs

```diff
@@ -9,33 +10,41 @@
 {-# LANGUAGE TypeOperators #-}
 
 module ExampleAPI
-  ( queryFile,
+  ( userAPI,
     server,
+    app,
+    Types,
   )
 where
 
-import Data.Aeson (ToJSON)
+import Data.Aeson (FromJSON (parseJSON), ToJSON (toJSON))
 import Data.Text (Text)
-import Elm.Ast (ElmDefinition)
+import Elm (elmStreetParseJson, elmStreetToJson)
 import Elm.Generic (Elm (..))
-import Elm.Print (showDoc)
 import GHC.Generics (Generic)
-import Prettyprinter (vsep)
-import Servant (Get, JSON, Proxy (..), Server, type (:<|>) (..), type (:>))
-import Servant.Foreign (HasForeignType (..))
-import ServantElm (elmForAPI)
+import Network.Wai (Application)
+import Servant (Get, JSON, Post, Proxy (..), ReqBody, Server, serve, type (:<|>) (..), type (:>))
 
 type UserAPI =
   "users" :> Get '[JSON] [User]
     :<|> "albert" :> Get '[JSON] User
+    :<|> "signup" :> ReqBody '[JSON] User :> Post '[JSON] User
+    :<|> "sth" :> Post '[JSON] User
 
 data User = User
-  { name :: String,
+  { name :: Text,
     age :: Int
   }
-  deriving (Generic, Elm)
+  deriving (Generic)
+  deriving anyclass (Elm)
 
-instance ToJSON User
+instance ToJSON User where toJSON = elmStreetToJson
+
+instance FromJSON User where parseJSON = elmStreetParseJson
```

Comment:
You can use `DerivingVia` instead, [as shown in the tutorial](https://github.com/Holmusk/elm-street#getting-started)